fission-web-server/library/Fission/Web/Server/Handler/User/ResendEmail.hs

```diff
@@ -0,0 +1,35 @@
+module Fission.Web.Server.Handler.User.ResendEmail (handler) where
+
+import           Servant
+
+import           Fission.Prelude
+
+import qualified Fission.Web.API.User.Email.Resend.Types      as API.Email
+
+import           Fission.Web.Server.Authorization.Types
+import qualified Fission.Web.Server.Challenge.Retriever.Class as Challenge
+import           Fission.Web.Server.Email.Class
+import           Fission.Web.Server.Email.Types
+import qualified Fission.Web.Server.Error                     as Web.Err
+import           Fission.Web.Server.Models
+
+
+handler ::
+  ( MonadThrow          m
+  , MonadLogger         m
+  , MonadEmail          m
+  , Challenge.Retriever m
+  )
+  => ServerT API.Email.Resend m
+
+handler Authorization { about = Entity _ User { userEmail = Nothing } } =
+  Web.Err.throw err422 { errBody = "There is no email associated with the user" }
+
+handler Authorization { about = Entity userId User { userUsername = username, userEmail = Just email } } = do
+  challenge <- Web.Err.ensureM $ Challenge.retrieve userId
+
+  sendVerificationEmail (Recipient email username) challenge >>= \case
+    Left _ ->
+      Web.Err.throw err500 { errBody = "Could not send verification email" }
+    Right _ ->
+      return NoContent
```

Comment:
Also, no need to change anything, but you may be able to save yourself some typing by using a single function head in this case. But literally doesn't matter 👍

Reply:
Also same thing here. Let's slap an `ensureM` on the `sendVerificationEmail`, and make sure that it has the correct error.

Reply:
Ideally we get something more specific than a 500 as well. But also realize that it's not always feasible, depending on what we get back from Send In Blue

Reply:
> Ideally we get something more specific than a 500 as well. But also realize that it's not always feasible, depending on what we get back from Send In Blue

Yeah, that's a good point. At least for now I've kept it possible to have the error depend on the `Servant.Client.ClientError` we get from the SIB request. I might take a look at the SIB API to maybe give better error messages.

Reply:
> Let's slap an ensureM on the sendVerificationEmail

FTR, did that :+1: